pkg/ratelimiter/concurrency_limiter.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ratelimiter
+
+import (
+	"sync"
+)
+
+type concurrencyLimiter struct {
+	mu      sync.RWMutex
+	current uint64
+	limit   uint64
+}
+
+func newConcurrencyLimiter(limit uint64) *concurrencyLimiter {
+	return &concurrencyLimiter{limit: limit}
+}
+
+func (l *concurrencyLimiter) allow() bool {
```

Comment:
maybe `acquire`, `allow` looks like a read-only method.

Reply:
followed [golang.org/x/time/rate](https://pkg.go.dev/golang.org/x/time/rate) Limiter

Reply:
BTW, we may need to implement the `Limit` interface if we are going to support gRPC. See https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/ratelimit/ratelimit.go#L11-L16

Reply:
> BTW, we may need to implement the `Limit` interface if we are going to support gRPC. See https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/ratelimit/ratelimit.go#L11-L16

IMO, this interface may not be suitable for our scene if our limit level is service